src/include/user/igo_nr.h

```diff
@@ -0,0 +1,63 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2021 Intelligo Corporation. All rights reserved.
+ *
+ * Author: Fu-Yun TSUO <fy.tsuo@intelli-go.com>
+ */
+
+#ifndef __USER_IGO_NR_H__
+#define __USER_IGO_NR_H__
+
+#include <stdint.h>
+
+#define SOF_IGO_NR_MAX_SIZE 4096		/* Max size for coef data in bytes */
+
+/*
+ * sof_igo_nr_config data[]
+
+ * int16_t input_channel_select[num_filters];  0 = ch0, 1 = 1ch1, ..
+ * int16_t output_channel_mix[num_filters];
+ * int16_t output_stream_mix[num_filters];
+ *
+ */
+
+#define IGO_PARAMS_MAJOR 1
+#define IGO_PARAMS_MINOR 0
+#define IGO_PARAMS_PATCH 0
+
+#define IGO_PARAMS_MAJOR_SHIFT 24
+#define IGO_PARAMS_MINOR_SHIFT 12
+#define IGO_PARAMS_PATCH_SHIFT 0
+#define IGO_PARAMS_VER(major, minor, patch) \
+						(((major) << IGO_PARAMS_MAJOR_SHIFT) | \
+						((minor) << IGO_PARAMS_MINOR_SHIFT) | \
+						((patch) << IGO_PARAMS_PATCH_SHIFT))
+#define IGO_PARAMS_VERSION IGO_PARAMS_VER(IGO_PARAMS_MAJOR, IGO_PARAMS_MINOR, IGO_PARAMS_PATCH)
+
+struct IGO_PARAMS {
+	uint32_t igo_params_ver;
+	uint32_t dump_data;
+	uint32_t nr_bypass;
+	uint32_t nr_mode1_en;
+	uint32_t nr_mode3_en;
+	uint32_t nr_ul_enable;
+	uint32_t agc_gain;
+	uint32_t nr_voice_str;
+	uint32_t nr_level;
+	uint32_t nr_mode1_floor;
+	uint32_t nr_mode1_od;
+	uint32_t nr_mode1_pp_param7;
+	uint32_t nr_mode1_pp_param8;
+	uint32_t nr_mode1_pp_param10;
+	uint32_t nr_mode3_floor;
+	uint32_t nr_mode1_pp_param53;
+};
```

Comment:
This should be packed if it's visible to userspace as it guarantees the ABI will align no matter the architecture.

Reply:
These parameters are visible to userspace. For example, the sof-ctl needs this structure to control igo_nr component.

Reply:
btw, packed was removed because of checkpatch warning.

Reply:
please ignore checkpatch warning here (we need to fix), the older XCC compiler does not understand the new syntax for ```packed``` and tends to produce errors. We need to use the old packed syntax (age just ignore the checkpatch errors).

Reply:
@lgirdwood  understand. thanks for explanation!